pyblish/plugin.py

```diff
@@ -1194,6 +1195,69 @@ def registered_targets():
     return list(_registered_targets)
 
 
+def register_discovery_filter(callback):
+    """Register a new plugin filter
+
+    Arguments:
+        callback (func): Function to execute on filter during discovery
+
+    Raises:
+        ValueError if `callback` is not callable.
+
+    """
+
+    if not hasattr(callback, "__call__"):
+        raise ValueError("%s is not callable" % callback)
+
+    _registered_plugin_filters.append(callback)
+
+
+def deregister_discovery_filter(callback):
+    """Deregister a plugin filter
+
+    Arguments:
+        callback (func): filtering function.
+
+    Raises:
+        ValueError on missing callback
+    """
+
+    _registered_plugin_filters.remove(callback)
+
+
+def deregister_all_discovery_filters():
+    """Deregisters all plugin filters"""
+
+    del _registered_plugin_filters[:]
+
+
+def registered_discovery_filters():
+    """Returns registered plugin filter callbacks"""
+
+    return _registered_plugin_filters
+
+
+def filter_plugins(plugins):
+    """Trigger registered plugin filters
+
+    Keyword arguments are passed from caller to callee.
+
+    Arguments:
+        plugins (Dict): dictionary of plugins to be filtered
+
+    """
+
+    if not _registered_plugin_filters:
+        return
+
+    filtered = False
+    for callback in _registered_plugin_filters:
+        try:
+            plugin, filtered = callback(plugins)
```

Comment:
There's no need for the callback to return anything; it can change the items in-place, like they do in your test below.